### PR TITLE
Update Questions Page

### PR DESCRIPTION
**Pull request by keaganpzh**
Let's update the questions page by:

- [x] Fetch question data from question service using HTTP
- [x] Create question modal that shows up when clicked
- [x] Update styling

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -14,10 +14,12 @@
     "@emotion/react": "^11.13.3",
     "@emotion/styled": "^11.13.0",
     "framer-motion": "^11.5.4",
+    "marked": "^14.1.2",
     "monaco-editor": "^0.52.0",
     "next": "14.2.10",
     "react": "^18",
-    "react-dom": "^18"
+    "react-dom": "^18",
+    "react-markdown": "^9.0.1"
```
Comment (XXJJXJ) [inline frontend/package.json:22]: I think somehow axios not imported for the frontend

File: package.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "dependencies": {
+    "axios": "^1.7.7"
+  }
+}
```
Comment (XXJJXJ) [inline package.json:5]: Oh whoops, it got added to the root directory 😂

**Review by XXJJXJ — COMMENTED** [review]
LGTM with small issues, some can be done in a different PR

1. Axios imported in the wrong directory
2. Gateway needs to enableCors too, once done, I think the other inner services might not need [TBC]
3. hooks to go through gateway, but for now idt it matters

File: package.json (hunk 2 above)
Comment (keaganpzh) [reply package.json]: thanks for the catch!

**Pull request merged**